src/sentry/static/sentry/app/components/confirm.tsx

```diff
@@ -135,149 +124,122 @@ class Confirm extends React.PureComponent<Props, State> {
-      return {
-        disableConfirmButton: props.disableConfirmButton,
-      };
-    }
-    return null;
+    return props.disableConfirmButton !== state.disableConfirmButton
+      ? {disableConfirmButton: props.disableConfirmButton}
+      : null;
   }
 
   confirming: boolean = false;
 
-  openModal = () => {
-    const {onConfirming, disableConfirmButton} = this.props;
-    if (typeof onConfirming === 'function') {
-      onConfirming();
-    }
-
-    this.setState({
-      isModalOpen: true,
-      disableConfirmButton: disableConfirmButton || false,
-    });
-
-    // always reset `confirming` when modal visibility changes
-    this.confirming = false;
-  };
-
-  closeModal = () => {
-    const {onCancel, disableConfirmButton} = this.props;
-    if (typeof onCancel === 'function') {
-      onCancel();
-    }
-    this.setState({
-      isModalOpen: false,
-      disableConfirmButton: disableConfirmButton || false,
-    });
-
-    // always reset `confirming` when modal visibility changes
-    this.confirming = false;
-  };
+  openModal = (e?: React.MouseEvent) => {
+    const {disabled, bypass, stopPropagation, onConfirm} = this.props;
 
-  handleConfirm = () => {
-    // `confirming` is used to make sure `onConfirm` is only called once
-    if (!this.confirming) {
-      this.props.onConfirm();
+    if (stopPropagation) {
+      e?.stopPropagation();
     }
 
-    // Close modal
-    this.setState({
-      isModalOpen: false,
-      disableConfirmButton: true,
-    });
-    this.confirming = true;
-  };
-
-  handleToggle = (e: React.MouseEvent): void => {
-    const {disabled, bypass, stopPropagation, onConfirm} = this.props;
     if (disabled) {
       return;
     }
 
-    if (e && stopPropagation) {
-      e.stopPropagation();
-    }
-
     if (bypass) {
       onConfirm();
       return;
     }
 
-    // Current state is closed, means it will toggle open
-    if (!this.state.isModalOpen) {
-      this.openModal();
-    } else {
-      this.closeModal();
-    }
+    const {onConfirming, disableConfirmButton} = this.props;
+
+    onConfirming?.();
+    this.setState({disableConfirmButton: disableConfirmButton || false});
+
+    // always reset `confirming` when modal visibility changes
+    this.confirming = false;
+
+    this.activateModal();
   };
 
-  render() {
-    const {
-      disabled,
-      message,
-      renderMessage,
-      priority,
-      confirmText,
-      cancelText,
-      children,
-      header,
-    } = this.props;
+  activateModal = () =>
+    openModal(({Header, Body, Footer, closeModal}) => {
+      const {
+        priority,
+        renderMessage,
+        message,
+        confirmText,
+        cancelText,
+        header,
+        onConfirm,
+        onCancel,
+        disableConfirmButton,
+      } = this.props;
+
+      const handleClose = () => {
+        onCancel?.();
+        this.setState({disableConfirmButton: disableConfirmButton || false});
```

Comment:
can `disableConfirmButton`  be something other than `boolean`?

Reply:
Yes it could be undefined. Changed it over to `??`.